### PR TITLE
CD-176157 Fix exception handling to not throw exceptions but call callback.error methods

### PR DESCRIPTION
**Pull request by johnwu96822**
Throwing exceptions will have side effects for the auth flow, so change the call error methods.

- [x] @johnny-lai 
- [x] @supriya

### CODE REVIEW

**Comment by johnny-lai** [comment]
lgtm

File: src/main/java/com/coupa/kafka/security/sasl/oauth/KafkaBrokerTokenCreator.java [hunk 1]
```diff
@@ -63,11 +63,12 @@ private void makeToken(OAuthBearerTokenCallback callback) throws IOException {
 
     if (tokenResponse == null) {
       LOGGER.error("Failed to get Sand token (token response is null)");
-      throw new IllegalArgumentException("Failed to get Sand token (token response is null)");
+      callback.error("invalid_client", "Failed to get Sand token (token response is null)", null);
```
Comment (johnny-lai) [inline src/main/java/com/coupa/kafka/security/sasl/oauth/KafkaBrokerTokenCreator.java]: How about `temporarily_unavailable`? This is from https://tools.ietf.org/html/rfc6749 Section 4.1.2.1

**Comment by supriya** [comment]
lgtm

**Comment by johnny-lai** [comment]
[lockit v4.9.2] merged via Lockit CLI on 2019-10-18 22:09:20.774167 -0700 PDT m=+5.969651788

**Pull request merged**